### PR TITLE
use SWR for data fetching

### PR DESCRIPTION
**Pull request by eatyourgreens**
Remove our homegrown `useEffect` hook and use [SWR](https://swr.vercel.app) to fetch results from Caesar.

### CODE REVIEW

**Review by shaunanoordin — APPROVED** [review]
## PR Overview

This PR adds the Stale-While-Revalidate library, which was previously used in [FEM](https://github.com/zooniverse/front-end-monorepo/pull/2677).

- New dependency: `swr`. Basically, 1. SWR is middleware in the 'fetch data' process. 2. whenever the window gains focus, SWR re-fetches the data. 3. if the data is different, it tells the component to update. If not, the component stays the same.
- The addition of this new library should not be felt on the user-side.

Code read looks good and tests are fine, though I need to note a new (minor) quirk detected in this PR. (I _think._ It's definitely in the `fetchAggregations()` trigger, so it _seems_ SWR-related, but I can't 100% rule out this randomly-triggered thing ain't happening on master.)

**New Quirk:** a redundant graphql fetch with no Subject ID is _sometimes_ triggered on page load
- Occasionally, when loading a Subject ([Galaxy Zoo for Schools example](https://localhost:3000/view/workflow/40/subject/475202?env=production)), there is an initial call to Caesar's graphql that's missing the subjectId parameter.
- In this screenshot, you can see two graphql calls in a row, when you only expect one:
    image
- The first (highlighted) graphql call has the payload: `query: "{\n    workflow(id: 40) {\n      reductions(subjectId: , reducerKey: \"init\") {\n        data\n      }\n      extracts(subjectId: , extractorKey: \"init\") {\n        data\n      }\n    }\n  }"` (note that subjectId is blank)
- The response is actually a 200 (surprise!) but with the body: `{"errors":[{"message":"Parse error on \":\" (COLON) at [3, 41]","locations":[{"line":3,"column":41}]}]}`
- The second graphql call is the correct call, with expected payload and response.

Basically, for some reason, subjectId is empty on page load, but only sometimes. 🤷 And then it quickly corrects itself. 

This quirk is NOT visible to the user. (Hence, it's not really a bug, as far as I'm concerned.)

### Testing

EDIT: forgot to add this in the initial post.

Tested on localhost with macOS + Chrome. [Same steps and successful results as with previous testing](https://github.com/zooniverse/zoo-notes/pull/89#pullrequestreview-963501983), albeit with https://localhost:3000 URL instead of http://localhost:3000

### Status

LGTM 👍 This PR gets an approval for me.

The new quirk doesn't seem serious enough to flag as a breaking problem, but I am curious to know what's up with that.

**Comment by shaunanoordin** [comment]
re: the Quirk :-

In SubjectPage's `const { subject, workflow } = usePanoptes(subjectId, workflowId) ; useCaesar(subject.id, workflow.id)`, could it be that usePanoptes() hook is able to return partial values? (workflow = {something}, subject = {empty}) But I thought it was designed to counter that, so the useCaesar() aggregation fetch is _only_ called after we've fetched _both_ Subject _and_ Workflow. 🤔

**Comment by eatyourgreens** [comment]
Oh, maybe the `useCaesar` hook only checks for a workflow ID before running the query. The subject and workflow requests run in parallel, so the code kind of assumes that the workflow will arrive after the subject.
